servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -122,9 +122,18 @@ public ReferencesResponse getAllReferences(ReferencesParams params) {
   }
 
   @Override
-  public Reference getReferenceByName(String refName) throws NessieNotFoundException {
+  public Reference getReferenceByName(GetReferenceParams params) throws NessieNotFoundException {
     try {
-      return makeRef(getStore().toRef(refName));
+      GetNamedRefsParams getNamedRefsParams =
+          params.isFetchAdditionalInfo()
+              ? GetNamedRefsParams.builder()
+                  .baseReference(BranchName.of(this.getConfig().getDefaultBranch()))
+                  .branchRetrieveOptions(RetrieveOptions.BASE_REFERENCE_RELATED_AND_COMMIT_META)
```

Comment:
Same here: add `.tagRetrieveOptions(RetrieveOptions.COMMIT_META)`

Unify the `GetNamedRefsParams` build in a utility method and use it from `getReferenceByName` + `getAllReferences`?

Reply:
done